lib/normalizer.c

```diff
@@ -1418,6 +1418,89 @@ grn_nfkc_normalize_unify_katakana_v_sounds(grn_ctx *ctx,
   return current;
 }
 
+static const unsigned char *
+grn_nfkc_normalize_unify_katakana_z_sounds(grn_ctx *ctx,
+                                           const unsigned char *start,
+                                           const unsigned char *current,
+                                           const unsigned char *end,
+                                           size_t *n_used_bytes,
+                                           size_t *n_used_characters,
+                                           unsigned char *unified_buffer,
+                                           size_t *n_unified_bytes,
+                                           size_t *n_unified_characters,
+                                           void *user_data)
+{
+  size_t char_length;
+  bool start_char;
+
+  char_length = (size_t)grn_charlen_(ctx, current, end, GRN_ENC_UTF8);
+  start_char = false;
+
+  *n_used_bytes = char_length;
+  *n_used_characters = 1;
+
+  if (char_length == 3) {
+    /* U+30C5 KATAKANA LETTER DU */
+    /* U+30BA KATAKANA LETTER ZU */
+    start_char = 
+      (current[0] == 0xe3 && current[1] == 0x83 && current[2] == 0x85) ||
+      (current[0] == 0xe3 && current[1] == 0x82 && current[2] == 0xba);
+  }
+
+  if (start_char) {
+    const unsigned char *next = current + char_length;
+    size_t next_char_length;
+
+    next_char_length = (size_t)grn_charlen_(ctx, next, end, GRN_ENC_UTF8);
+    if (next_char_length == 3 &&
+        next[0] == 0xe3 &&
+        next[1] == 0x82) {
+      if (next[2] == 0xa1) { /* U+30A1 KATAKANA LETTER SMALL A */
+        /* U+30B6 KATAKANA LETTER ZA */
+        unified_buffer[(*n_unified_bytes)++] = current[0];
```

Comment:
Or directly specify `0xe3`.
I have use `current[0]` because current implementation use `current` as possible as we can.

Reply:
I think that `current[0]` is better because existing implementation already uses the style.

Reply:
Thanks.
I see.